# Settings of the plugins. (The plugins should be stored in script/plugins)

# min_filter fills in minimum value around the invalid cell.
min_filter:                                   
  enable: True                                # weather to load this plugin
  fill_nan: False                             # Fill nans to invalid cells of elevation layer.
  is_height_layer: True                       # If this is a height layer (such as elevation) or not (such as traversability)
  layer_name: "min_filter"                    # The layer name.
  extra_params:                               # This params are passed to the plugin class on initialization.
    dilation_size: 1                         # The patch size to apply
    iteration_n: 30                           # The number of iterations
# Apply smoothing.
smooth_filter:                                
  enable: True
  fill_nan: False
  is_height_layer: True
  layer_name: "smooth"
  extra_params:
    input_layer_name: "min_filter"
# Apply inpainting using opencv
inpainting:
  enable: True
  fill_nan: False
  is_height_layer: True
  layer_name: "inpaint"
  extra_params:
    method: "telea"                           # telea or ns
# Apply smoothing for inpainted layer

# robot_centric_elevation:                                  # Use the same name as your file name.
# #  type: "robot_centric_elevation"
#   enable: False                                # weather to load this plugin
#   fill_nan: False                              # Fill nans to invalid cells of elevation layer.
#   is_height_layer: True                       # If this is a height layer (such as elevation) or not (such as traversability)
#   layer_name: "robot_centric_elevation"                 # The layer name.
#   extra_params:                               # This params are passed to the plugin class on initialization.
# #    add_value: 2.0                            # Example param
#     resolution: 0.04
#     threshold: 1.1
#     use_threshold: True

semantic_filter:
  type: "semantic_filter"
  enable: True
  fill_nan: False
  is_height_layer: False
  layer_name: "sem_fil"
  extra_params:
    classes: ['grass','tree','fence','dirt']

traversable_fusion:                                      # Name of your filter
  type: "traversable_fusion"                             # Specify the name of your plugin (the name of your file name).
  enable: True                                # weather to load this plugin
  fill_nan: True                              # Fill nans to invalid cells of elevation layer.
  is_height_layer: False                       # If this is a height layer (such as elevation) or not (such as traversability)
  layer_name: "traversable_fusion_layer"                 # The layer name.
  extra_params:                               # This params are passed to the plugin class on initialization.
    geo_trav_weight: 0.5                           # Example param

traversable_fusion_sem:                                      # Name of your filter
  type: "traversable_fusion"                             # Specify the name of your plugin (the name of your file name).
  enable: True                                # weather to load this plugin
  fill_nan: True                              # Fill nans to invalid cells of elevation layer.
  is_height_layer: False                       # If this is a height layer (such as elevation) or not (such as traversability)
  layer_name: "traversable_fusion_sem_layer"                 # The layer name.
  extra_params:                               # This params are passed to the plugin class on initialization.
    geo_trav_weight: 0.0                      # Example param

traversable_fusion_min:                                      # Name of your filter
  type: "min_fusion"                             # Specify the name of your plugin (the name of your file name).
  enable: True                                # weather to load this plugin
  fill_nan: True                              # Fill nans to invalid cells of elevation layer.
  is_height_layer: False                       # If this is a height layer (such as elevation) or not (such as traversability)
  layer_name: "traversable_fusion_min_layer"                 # The layer name.
  extra_params:                               # This params are passed to the plugin class on initialization.
    geo_trav_weight: 0.5                      # Example param
    min_safe_geo: 0.3                             # Example param
    min_safe_sem: 0.1

info_goal:
  type: "info_goal"
  enable: True
  fill_nan: True
  is_height_layer: False
  layer_name: "info_goal"
  extra_params:
    robot_size: 1
    robot_cell_sight: 100
    
# semantic_traversability:
#   type: "semantic_traversability"
#   enable: False
#   fill_nan: False
#   is_height_layer: False
#   layer_name: "sem_traversability"
#   extra_params:
#     layers: ['traversability','robot_centric_elevation']
#     thresholds: [0.3,0.5]
#     type: ['traversability', 'elevation']

